handler/graphql.go

```diff
@@ -250,6 +251,14 @@ func (tw *tracerWrapper) EndOperationExecution(ctx context.Context) {
 	tw.tracer1.EndOperationExecution(ctx)
 }
 
+// WebsocketOnInitFunc is called when the server receives connection init message from the client.
+// This can be used to check initial payload to see whether to accept the websocket connection.
+func WebsocketOnInitFunc(websocketOnInitFunc func(ctx context.Context, initPayload InitPayload) bool) Option {
```

Comment:
We will not support custom errors, and only return true or false on initialise failure?
```suggestion
func WebsocketOnInitFunc(websocketOnInitFunc func(ctx context.Context, initPayload InitPayload) (bool, error)) Option {
```

Reply:
_hmm ..._ I guess it depends on `func (c *wsConnection) init() bool` and whether or not you send the custom error message via `c.sendConnectionError`; like with [`c.sendConnectionError("invalid init payload")`](https://github.com/99designs/gqlgen/pull/750/files#diff-857d352c30d3f33e4fed201e953d953cR100)

Reply:
I didn't want to give this too many options. I don't think the server needs to tell the client why it fails either. If there's a need for it we can create another PR to follow up

Reply:
Maybe just `func(ctx context.Context, initPayload InitPayload) error`?